example-application/error-handling.js

```diff
@@ -0,0 +1,41 @@
+const mailer = require("./libraries/mailer");
+
+// This file simulates real-world error handler that makes this component observable
+const errorHandler = {
+  handleError: async (errorToHandle) => {
+    console.error(errorToHandle);
+    metricsExporter.fireMetric("error", { errorName: errorToHandle.name || "generic-error" });
+    // This is used to simulate sending email to admin when an error occurs
+    // In real world - The right flow is sending alerts from the monitoring system
+    mailer.send("Error occured", errorToHandle, "admin@our-domain.io");
+
+    // A common best practice is to crash when an unknown error (non-trusted) is being thrown
+    decideWhetherToCrash(errorToHandle);
+  },
+};
+
+const decideWhetherToCrash = (error) => {
+  if (!error.isTrusted) {
```

Comment:
If I understand the code correctly, this will crash our app: `throw 1;`

Reply:
@mikicho Yes, unless you throw 1 from a inside a request handler. If you throw 1 on process start - Have to assume that something unknown and terrible happened

Reply:
> unless you throw 1 from a inside a request handler

@goldbergyoni are you sure?
we don't wrap errors from the routers..


```
 expressApp.use(async (err, req, res, next) => {
    if (typeof err === "object") {
      if (err.isTrusted === undefined || err.isTrusted === null) {
        err.isTrusted = true; //Error during a specific request is usually not catastrophic and should not lead to process exit
      }
    }
    await errorHandler.handleError(err);
    res.status(500).end();
  });
```

Reply:
@mikicho I added a test for this flow now - Throwing unknown error during a request, the process stays alive. The post route has try-catch.